### ISSUE TITLE
CKV2_AWS_23 warns if alias has data resources

### ISSUE THREAD

**Issue by dullest** [bridgecrewio/checkov#1373]
**Describe the bug**
This is similar to https://github.com/bridgecrewio/checkov/issues/1359
`aws_route53_record` has data as alias in this case.
Is this warned by design of `CKV2_AWS_23` rule?

```
Check: CKV2_AWS_23: "Route53 A Record has Attached Resource"
	FAILED for resource: aws_route53_record.r53
	File: /r53.tf:10-20

		10 | resource "aws_route53_record" "r53" {
		11 |   zone_id = "${data.aws_route53_zone.zone.zone_id}"
		12 |   name    = "${var.dummy_hostname}"
		13 |   type    = "A"
		14 | 
		15 |   alias {
		16 |     name                   = "${data.aws_lb.lb_example.dns_name}"
		17 |     zone_id                = "${data.aws_lb.lb_example.zone_id}"
		18 |     evaluate_target_health = true
		19 |   }
		20 | }
```

**To Reproduce**
1. Create a zone and a LB  by aws console or separated terraform codes.
2. Create `data`(refers a zone and a LB what were created by step1) and `aws_route53_record` resource what has alias for the `data`
```terraform
variable "example_lb_arn" {}
variable "example_zone_id" {}

data "aws_lb" "example" {
  arn = var.example_lb_arn
}

data "aws_route53_zone" "example" {
  zone_id      = var.example_zone_id
  private_zone = true
}

resource "aws_route53_record" "example" {
  zone_id = data.aws_route53_zone.example.zone_id
  name    = "example"
  type    = "A"

  alias {
    name                   = data.aws_lb.example.dns_name
    zone_id                = data.aws_lb.example.zone_id
    evaluate_target_health = true
  }
}
```
3. See error

**Expected behavior**
Able to reference data without getting a failed warning thrown.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Alpine Linux (`bridgecrew/checkov:2.0.266` from Docker Hub)
 - Checkov Version [e.g. 22]: 2.0.266 

**Additional context**
Add any other context about the problem here (e.g. code snippets).

**Comment by JamesWoolfenden** [comment]
this is just awaiting https://github.com/bridgecrewio/checkov/pull/1375 #1375

**Comment by dullest** [comment]
Thank you for the fix.
